### PR TITLE
Avoid amdllpc assert with filename containing '*' on non-Windows

### PR DESCRIPTION
**Pull request by trenouf**
Change-Id: I400ebd06d616dd975f5f6de384277de5ed65969b

### CODE REVIEW

File: tool/amdllpc.cpp [hunk 1]
```diff
@@ -1417,17 +1417,6 @@ void FindAllMatchFiles(
     FindClose(pSearchHandle);
     return;
 }
-#else
```
Comment (linqun) [inline tool/amdllpc.cpp]: How to hit this case? actually, I implemented the real implmentation on linux when I add this function.
But when I run the test on linux, I found linux shell expand all * and ?, I replace it with an assert.

Comment (trenouf) [reply tool/amdllpc.cpp]: 1. You can hit it by saying amdllpc "*.pipe" (i.e. quote the * so it does not get expanded).
2. If you do not quote the * but the shell finds no matching file for the glob *.pipe, the default setting of the shell is to pass the *.pipe verbatim through to the command, so the command will say "*.pipe: file not found".

Comment (trenouf) [reply tool/amdllpc.cpp]: In (2), that was supposed to say:
the default setting of the shell is to pass the `*.pipe` verbatim through to the command

Pesky markdown mangled my comment. :-)

**Comment by amdvlk-admin** [comment]
### Test summary for commit dcb569f0a70aca9efbb0f347cfb119ebf7de4749
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 7438ae21a2e8e3c782f82745249da539a19c193e</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit b1e22fa3308100dca8211742be33b887953920c5
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 7438ae21a2e8e3c782f82745249da539a19c193e</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**